### ISSUE TITLE
301 gets :cognitect.anomlies/fault

### ISSUE THREAD

**Issue by dchelimsky** [cognitect-labs/aws-api#237]
## Dependencies

``` clojure
{:deps {com.cognitect.aws/api       {:mvn/version "0.8.664"}
        com.cognitect.aws/endpoints {:mvn/version "1.1.12.456"}
        com.cognitect.aws/s3        {:mvn/version "847.2.1365.0"}}}
```

## Description with failing test case

Release 0.8.664 added status and headers when we get a 301 (along with other non-2xx's), so users can detect the 301 and do the right thing. One thing that would improve this slightly is returning an anomaly/category of `incorrect` instead of `fault` (which is a catch-all). This would be more precise.

**Comment by dchelimsky** [comment]
Changed to `:cognitect.anomalies/incorrect` in 0.8.666

**Comment by dchelimsky** [comment]
I thought I'd fixed this, but later learned that, due to the fat AWS does not provide a Location header with a 301, the jetty client is throwing an exception which pushes control through a different part of the code. Investigating.

**Comment by dchelimsky** [comment]
Fixed in 0.8.673 (it was fixed in ~~0.8.670~~, but that had a perf warning, so don't use that one)
